toolchest_client/files/general.py

```diff
@@ -30,19 +32,31 @@ def assert_exists(path, must_be_file=False, must_be_directory=False):
         raise ValueError(f"Directory entry at {path} is not a directory")
 
 
-def check_file_size(file_path, max_size_bytes=None):
+def check_file_size(file_path, max_size_bytes=None, is_s3_uri=False):
     """Raises an error if the file is above the non-multipart upload limit for S3 (5GB)
 
     :param file_path: A path to a file.
     :type file_path: string
     :param max_size_bytes: Maximum number of bytes allowed for a file. Throws error if above limit.
     :type max_size_bytes: int | None
     """
-    assert_exists(file_path, must_be_file=True)
-    file_size_bytes = os.stat(file_path).st_size
+    if not is_s3_uri:
+        assert_exists(file_path, must_be_file=True)
+        file_size_bytes = os.stat(file_path).st_size
+    else:
+        # Get file size S3 metadata, via boto3.
+        s3_file_params = get_params_from_s3_uri(file_path)
+        s3_client = boto3.client("s3")
+        response = s3_client.head_object(
+            Bucket=s3_file_params["bucket"],
+            Key=s3_file_params["key"],
+        )
+        file_size_bytes = response["ContentLength"]
```

Comment:
This check be better suited for the API, where we actually can enforce data sizes (barring a race condition exploit where the file is replaced on S3 after a size check).

I'm fine with this existing here, but worth noting.